Use string literals for transaction category localized messages

Gudahtt opened the pull request:
We now use string literals for all transaction category localized messages. This makes it easier to verify that we have translations for each of them, and that we aren't leaving any unused translations around.

Gudahtt committed:
Use string literals for transaction category localized messages

We now use string literals for all transaction category localized
messages. This makes it easier to verify that we have translations for
each of them, and that we aren't leaving any unused translations around.

jgallegos91 reviewed (dismissed):
I like the idea

kumavis commented on ui/app/helpers/utils/transactions.util.js:
```diff
@@ -205,3 +205,53 @@ export function getBlockExplorerUrlForTx(networkId, hash, rpcPrefs = {}) {
+        return defaultTitle;
+      }
+      throw new Error(
+        `Unrecognized transaction category: ${transactionCategory}`,
```
maybe weird that we dont hit this validation if the defaultTitle is present

Gudahtt replied:
I meant the default case to imply that it was OK for the category to not be matched. That's how it's used in `useTransactionDisplayData` for example - there was a fallback present.

But now that you mention it... that doesn't seal ideal, I agree. Maybe we can remove the need for that fallback case. I'll look into that.

Gudahtt replied:
This was addressed in https://github.com/MetaMask/metamask-extension/pull/10391/commits/a8ab8e8f99ca273a9117b93f1e9348e7add71952

The default case has been replaced by another condition in the one case it was relied upon.

Gudahtt committed:
Remove default case for `getTransactionCategoryTitle`

The default case of `getTransactionCategoryTitle` wasn't necessary. I
audited all cases where the `transactionCategory` property is assigned,
and they all assign it to known transaction categories. So it should be
impossible for that to be an unrecognized value.

The only real case the fallback was useful for was when `null` or
`undefined` was given as the transaction category in the confirm
transaction base component. That case was handled there instead.

brad-decker reviewed (approved):
LGTM.
